### ISSUE TITLE
v22.11.1 crashes with disk I/O error / Optimistic lock on the database failed

### ISSUE THREAD

**Issue by BTCBellyButton** [ElementsProject/lightning#5984]
### Issue and Steps to Reproduce
After upgrading to v22.11.1 and updating the DB  

After awhile I get this error and lightningd restarts.
```
2023-02-08T22:48:57.573Z **BROKEN** lightningd: Error executing statement: wallet/wallet.c:3860: UPDATE utxoset SET spendheight = ? WHERE txid = ? AND outnum = ?: disk I/O error

2023-02-08T22:48:57.708Z **BROKEN** lightningd: FATAL SIGNAL 6 (version 22.11.1)
2023-02-08T22:48:57.708Z **BROKEN** lightningd: backtrace: common/daemon.c:38 (send_backtrace) 0x564f7b44f13b
2023-02-08T22:48:57.708Z **BROKEN** lightningd: backtrace: common/daemon.c:46 (crashdump) 0x564f7b44f18d
2023-02-08T22:48:57.708Z **BROKEN** lightningd: backtrace: /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/x86_64/sigaction.c:0 ((null)) 0x7f3f001d408f
2023-02-08T22:48:57.708Z **BROKEN** lightningd: backtrace: ../sysdeps/unix/sysv/linux/raise.c:51 (__GI_raise) 0x7f3f001d400b
2023-02-08T22:48:57.708Z **BROKEN** lightningd: backtrace: /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:79 (__GI_abort) 0x7f3f001b3858
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: lightningd/log.c:854 (fatal_vfmt) 0x564f7b40c4c5
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: wallet/wallet.c:61 (db_fatal) 0x564f7b439ba4
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: db/utils.c:182 (db_exec_prepared_v2) 0x564f7b496466
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: wallet/wallet.c:3868 (wallet_outpoint_spend) 0x564f7b43e15a
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: lightningd/chaintopology.c:678 (topo_update_spends) 0x564f7b3ee895
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: lightningd/chaintopology.c:729 (add_tip) 0x564f7b3eebe2
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: lightningd/chaintopology.c:823 (get_new_block) 0x564f7b3eed03
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: lightningd/bitcoind.c:383 (getrawblockbyheight_callback) 0x564f7b3ecb75
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: lightningd/plugin.c:585 (plugin_response_handle) 0x564f7b4273ac
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: lightningd/plugin.c:696 (plugin_read_json_one) 0x564f7b42a40a
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: lightningd/plugin.c:741 (plugin_read_json) 0x564f7b42a554
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:59 (next_plan) 0x564f7b4f5146
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:407 (do_plan) 0x564f7b4f5613
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:417 (io_ready) 0x564f7b4f56b0
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: ccan/ccan/io/poll.c:453 (io_loop) 0x564f7b4f68ec
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: lightningd/io_loop_with_timers.c:22 (io_loop_with_timers) 0x564f7b405dba
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: lightningd/lightningd.c:1201 (main) 0x564f7b40a0a9
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: ../csu/libc-start.c:308 (__libc_start_main) 0x7f3f001b5082
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0x564f7b3ec0ad
2023-02-08T22:48:57.709Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0xffffffffffffffff




2023-02-08T22:50:10.477Z INFO    plugin-bcli: bitcoin-cli initialized and connected to bitcoind.
2023-02-08T22:50:20.360Z UNUSUAL plugin-plugin.js: --- Starting the cl-rest server ---
2023-02-08T22:50:20.372Z INFO    lightningd: --------------------------------------------------
...
...

```

### `getinfo` output
```
{
   "id": "",
   "alias": "",
   "color": "0228cc",
   "num_peers": 10,
   "num_pending_channels": 0,
   "num_active_channels": 10,
   "num_inactive_channels": 0,
   "address": [
      {
         "type": "ipv4",
         "address": "",
         "port": 9735
      },
      {
         "type": "torv3",
         "address": "...onion",
         "port": 9735
      }
   ],
   "binding": [
      {
         "type": "ipv6",
         "address": "::",
         "port": 9735
      },
      {
         "type": "ipv4",
         "address": "0.0.0.0",
         "port": 9735
      }
   ],
   "version": "22.11.1",
   "blockheight": 775640,
   "network": "bitcoin",
   "msatoshi_fees_collected": ...,
   "fees_collected_msat": "...",
   "lightning-dir": "/mnt/md0/C-Lightning/bitcoin",
   "our_features": {
      "init": "08a000080a69a2",
      "node": "88a000080a69a2",
      "channel": "",
      "invoice": "02000000024100"
   }
}

```

TLDR: https://github.com/ElementsProject/lightning/issues/5984#issuecomment-1428882523

**Comment by vincenzopalazzo** [comment]
The IO error do not sounds depending from cln?

I read this https://stackoverflow.com/questions/9993555/disk-i-o-error-with-sqlite maybe if you ran some of the tests mentioned there we can help to diagnose the problem?

**Comment by BTCBellyButton** [comment]
I'll take a look.
In the meantime an update.
After upgrading the DB I did NOT reboot my machine.
So after seeing 3 similar crashes (1 a couple of minutes after the plugin Circular did some work, the second after quite some time after using the other plugin Rebalance, and finally the last one without running anything else than clightning-REST), I checked my RAID for issues but it had none.
I eventually decided to reboot (only with REST active) and since then, no crashes.
I'll try starting Circular and seeing if it crashes again.
So maybe, after the upgrade something needed a machine reboot?

**Comment by BTCBellyButton** [comment]
OK. Update.
After about 13hours with no problems I manually started the Circular Plugin and few minutes after its first GraphRefresh operation it crashed like yesterday.

```

2023-02-09T15:19:02.893Z INFO    plugin-circular: node.go:111 setOptions: initializing node
2023-02-09T15:19:05.225Z INFO    plugin-circular: file.go:41 LoadGraphFromFile: graph loaded successfully
2023-02-09T15:19:05.226Z INFO    plugin-circular: cron.go:46 refreshGraph: refreshing graph
2023-02-09T15:19:12.320Z INFO    plugin-circular: cron.go:74 refreshGraph: graph has been refreshed
2023-02-09T15:19:12.853Z INFO    plugin-circular: node.go:92 Init: node initialized
2023-02-09T15:19:12.854Z INFO    plugin-circular: 10:19:12 main.go:39: circular successfully init'd!
2023-02-09T15:20:07.500Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-chan#696: Peer transient failure in CHANNELD_NORMAL: Disconnected
2023-02-09T15:20:11.172Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-chan#696: Peer transient failure in CHANNELD_NORMAL: Disconnected
2023-02-09T15:20:16.348Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-chan#696: Peer transient failure in CHANNELD_NORMAL: Disconnected
2023-02-09T15:20:21.861Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-channeld-chan#696: Peer connection lost
2023-02-09T15:20:21.955Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-chan#696: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
2023-02-09T15:20:25.451Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-chan#696: Peer transient failure in CHANNELD_NORMAL: Disconnected
2023-02-09T15:20:30.966Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-channeld-chan#696: Peer connection lost
2023-02-09T15:20:30.967Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-chan#696: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
2023-02-09T15:20:42.625Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-chan#696: Peer transient failure in CHANNELD_NORMAL: Disconnected
2023-02-09T15:20:48.142Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-channeld-chan#696: Peer connection lost
2023-02-09T15:20:48.142Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-chan#696: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
2023-02-09T15:21:16.439Z INFO    037659a0ac8eb3b8d0a720114efc861d3a940382dcfa1403746b4f8f6b2e8810ba-chan#696: Peer transient failure in CHANNELD_NORMAL: Disconnected
2023-02-09T15:27:57.557Z INFO    027ce055380348d7812d2ae7745701c9f93e70c1adeb2657f053f91df4f2843c71-chan#701: Peer transient failure in CHANNELD_NORMAL: Disconnected
2023-02-09T15:29:12.001Z INFO    plugin-circular: cron.go:46 refreshGraph: refreshing graph
2023-02-09T15:29:12.042Z INFO    plugin-circular: cron.go:101 refreshLiquidity: liquidity has been reset on 144394 channels
2023-02-09T15:29:18.888Z INFO    plugin-circular: cron.go:74 refreshGraph: graph has been refreshed
2023-02-09T15:29:53.807Z INFO    027ce055380348d7812d2ae7745701c9f93e70c1adeb2657f053f91df4f2843c71-channeld-chan#701: Peer connection lost
2023-02-09T15:29:53.808Z INFO    027ce055380348d7812d2ae7745701c9f93e70c1adeb2657f053f91df4f2843c71-chan#701: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
2023-02-09T15:29:53.912Z INFO    027ce055380348d7812d2ae7745701c9f93e70c1adeb2657f053f91df4f2843c71-chan#701: Peer transient failure in CHANNELD_NORMAL: Disconnected
2023-02-09T15:30:37.046Z INFO    03bb88ccc444534da7b5b64b4f7b15e1eccb18e102db0e400d4b9cfe93763aa26d-chan#625: Peer transient failure in CHANNELD_NORMAL: Disconnected
2023-02-09T15:35:13.868Z INFO    027ce055380348d7812d2ae7745701c9f93e70c1adeb2657f053f91df4f2843c71-channeld-chan#701: Peer connection lost
2023-02-09T15:35:13.869Z INFO    027ce055380348d7812d2ae7745701c9f93e70c1adeb2657f053f91df4f2843c71-chan#701: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
2023-02-09T15:35:17.847Z INFO    027ce055380348d7812d2ae7745701c9f93e70c1adeb2657f053f91df4f2843c71-chan#701: Peer transient failure in CHANNELD_NORMAL: Disconnected

2023-02-09T15:36:00.427Z **BROKEN** lightningd: Error executing statement: wallet/wallet.c:3860: UPDATE utxoset SET spendheight = ? WHERE txid = ? AND outnum = ?: disk I/O error

2023-02-09T15:36:01.189Z **BROKEN** lightningd: FATAL SIGNAL 6 (version 22.11.1)
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: common/daemon.c:38 (send_backtrace) 0x556d505d513b
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: common/daemon.c:46 (crashdump) 0x556d505d518d
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/x86_64/sigaction.c:0 ((null)) 0x7f78fa30708f
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: ../sysdeps/unix/sysv/linux/raise.c:51 (__GI_raise) 0x7f78fa30700b
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:79 (__GI_abort) 0x7f78fa2e6858
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: lightningd/log.c:854 (fatal_vfmt) 0x556d505924c5
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: wallet/wallet.c:61 (db_fatal) 0x556d505bfba4
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: db/utils.c:182 (db_exec_prepared_v2) 0x556d5061c466
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: wallet/wallet.c:3868 (wallet_outpoint_spend) 0x556d505c415a
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: lightningd/chaintopology.c:678 (topo_update_spends) 0x556d50574895
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: lightningd/chaintopology.c:729 (add_tip) 0x556d50574be2
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: lightningd/chaintopology.c:823 (get_new_block) 0x556d50574d03
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: lightningd/bitcoind.c:383 (getrawblockbyheight_callback) 0x556d50572b75
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: lightningd/plugin.c:585 (plugin_response_handle) 0x556d505ad3ac
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: lightningd/plugin.c:696 (plugin_read_json_one) 0x556d505b040a
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: lightningd/plugin.c:741 (plugin_read_json) 0x556d505b0554
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:59 (next_plan) 0x556d5067b146
2023-02-09T15:36:01.189Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:407 (do_plan) 0x556d5067b613
2023-02-09T15:36:01.190Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:417 (io_ready) 0x556d5067b6b0
2023-02-09T15:36:01.190Z **BROKEN** lightningd: backtrace: ccan/ccan/io/poll.c:453 (io_loop) 0x556d5067c8ec
2023-02-09T15:36:01.190Z **BROKEN** lightningd: backtrace: lightningd/io_loop_with_timers.c:22 (io_loop_with_timers) 0x556d5058bdba
2023-02-09T15:36:01.190Z **BROKEN** lightningd: backtrace: lightningd/lightningd.c:1201 (main) 0x556d505900a9
2023-02-09T15:36:01.190Z **BROKEN** lightningd: backtrace: ../csu/libc-start.c:308 (__libc_start_main) 0x7f78fa2e8082
2023-02-09T15:36:01.190Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0x556d505720ad
2023-02-09T15:36:01.190Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0xffffffffffffffff


```

**Comment by grubles** [comment]
Have you checked `dmesg`? There may be a hardware issue.

**Comment by BTCBellyButton** [comment]
> Have you checked `dmesg`? There may be a hardware issue.

Nothing wrong there, apparently...

**Comment by BTCBellyButton** [comment]
From the crash log:

```
+233.880039990 lightningdDEBUG: Adding block 775761: 00000000000000000004304afb2ae1b19faaab27b852b1dda8b52a00ec8c56e8
+259.554759064 lightningdBROKEN: Error executing statement: wallet/wallet.c:3860: UPDATE utxoset SET spendheight = ? WHERE txid = ? AND outnum = ?: disk I/O error
+260.189493170 lightningdBROKEN: FATAL SIGNAL 6 (version 22.11.1)

```

**Comment by BTCBellyButton** [comment]
It's getting worse. Even after a reboot.
![image](https://user-images.githubusercontent.com/68573637/217895171-09d737f2-1dcc-45a0-b615-9f3193fd4f37.png)


Now I also removed the c-lightning-REST plugin from the config file.

**Comment by grubles** [comment]
Is your sqlite database using the defaults? You don't have write-ahead logging enabled?

**Comment by BTCBellyButton** [comment]
> Is your sqlite database using the defaults? You don't have write-ahead logging enabled?

Yes, never messed with it.

lightnind.service:

```
[Unit]
Description=Core Lightning daemon
Requires=bitcoind.service
After=bitcoind.service
Wants=network-online.target
After=network-online.target

[Service]
#ExecStart=/usr/bin/lightningd --daemon --conf /etc/lightningd/lightningd.conf --pid-file=/run/lightningd/lightningd.pid
ExecStart=/usr/bin/lightningd --lightning-dir=/mnt/md0/C-Lightning/ --daemon --log-file=/mnt/md0/C-Lightning/log --pid-file=/run/lightningd/lightningd.pid

# Creates /run/lightningd owned by bitcoin
RuntimeDirectory=lightningd

User=go
Group=go
Type=forking
PIDFile=/run/lightningd/lightningd.pid
Restart=on-failure
RestartSec=5s
StartLimitInterval=10

# Hardening measures
####################

# Provide a private /tmp and /var/tmp.
PrivateTmp=true

# Mount /usr, /boot/ and /etc read-only for the process.
ProtectSystem=full

# Disallow the process and all of its children to gain
# new privileges through execve().
NoNewPrivileges=true

# Use a new /dev namespace only populated with API pseudo devices
# such as /dev/null, /dev/zero and /dev/random.
PrivateDevices=true

# Deny the creation of writable and executable memory mappings.
#MemoryDenyWriteExecute=true

[Install]
WantedBy=multi-user.target

```

Config:
```
#Lightning node customization options
alias=BBB
fee-base=0
fee-per-satoshi=0
min-capacity-sat=2000000
funding-confirms=4
network=bitcoin
large-channels
#lightning-dir=/mnt/md0/C-Lightning/

#log-level=debug
#log-level=broken
#log-level=unusual
#log-level=info

#Lightning channel and HTLC options
max-concurrent-htlcs=200
autocleaninvoice-cycle=1800
autocleaninvoice-expired-by=3600

#Networking Options
bind-addr=
announce-addr=....:9735
announce-addr=....onion:9735
proxy=127.0.0.1:9050

#Plugins
#plugin=/home/go/c-lightning-REST/plugin.js
#rest-port=3001
#rest-docport=4001
#rest-protocol=https
#plugin=/mnt/md0/C-Lightning/plugins.bak/circular/circular

```

**Comment by BTCBellyButton** [comment]
Is it possible that the 3rd parties plugin mechanism isn't working correctly with the upgraded DB?

**Comment by BTCBellyButton** [comment]
Kind of worried here...

**Comment by BTCBellyButton** [comment]
> You don't have write-ahead logging enabled?

As I said I never messed with the DB but there are this 2 files:
```
-rw-r--r-- 1 go go  64K Feb  9 23:47 lightningd.sqlite3-shm
-rw-r--r-- 1 go go  24M Feb  9 23:47 lightningd.sqlite3-wal

```
If they should not be there, how do I go about it? I have 0 experience with SQLite...

**Comment by BTCBellyButton** [comment]
UPDATE
So far it seemed that the problem was somehow connected with the 3rd party plugins mechanism because it crashed around plugins' events. 
I always started lightningd via Systemd. 
But after the first crash in 6 hours with no 3d party plugin running it kept crushing after a couple of minute at every attempt to restart the service.
So last night I started it manually simply with `lightningd --lightning-dir=/mnt/md0/C-Lightning/` AND at the moment it's still running with no crashes.

Does this light any bulbs?

**Comment by cdecker** [comment]
Yes, you just created a completely new node with a new DB, that isn't affected by any disk corruption you had in the original.

The flaky behavior is likely due to only a few utxoset pages being on bad sectors, causing a crash whenever they get touched, but not in a reproducible manner. The plugin's idea is likely just a fluke, since plugin's do not have access to the DB file generally speaking.

**Comment by BTCBellyButton** [comment]
> Yes, you just created a completely new node with a new DB, 

Not sure I follow. I am using the same directory and config file as before. I just started from command line instead of systemd.

**Comment by BTCBellyButton** [comment]
And after about 20 hours with no crashes I eventually started c-lightning-REST and Circular plugins it crashed again.
From Command Line there are quite few more info:

```
2023-02-10T21:48:42.017Z INFO    plugin-circular: cron.go:101 refreshLiquidity: liquidity has been reset on 6 channels

Optimistic lock on the database failed. There may be a concurrent access to the database. Aborting since concurrent access is unsafe.
2023-02-10T21:48:45.235Z **BROKEN** lightningd: Optimistic lock on the database failed. There may be a concurrent access to the database. Aborting since concurrent access is unsafe.
lightningd: FATAL SIGNAL 6 (version 22.11.1)
0x55c0fc72c0f3 send_backtrace
	common/daemon.c:33
0x55c0fc72c18d crashdump
	common/daemon.c:46
0x7f691702508f ???
	/build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/x86_64/sigaction.c:0
0x7f691702500b __GI_raise
	../sysdeps/unix/sysv/linux/raise.c:51
0x7f6917004858 __GI_abort
	/build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:79
0x55c0fc6e94c5 fatal_vfmt
	lightningd/log.c:854
0x55c0fc716ba4 db_fatal
	wallet/wallet.c:61
0x55c0fc74131d db_data_version_incr
	db/exec.c:114
0x55c0fc7415bb db_commit_transaction
	db/exec.c:151
0x55c0fc70bda2 sd_msg_read
	lightningd/subd.c:585
0x55c0fc7d2146 next_plan
	ccan/ccan/io/io.c:59
0x55c0fc7d2613 do_plan
	ccan/ccan/io/io.c:407
0x55c0fc7d26b0 io_ready
	ccan/ccan/io/io.c:417
0x55c0fc7d38ec io_loop
	ccan/ccan/io/poll.c:453
0x55c0fc6e2dba io_loop_with_timers
	lightningd/io_loop_with_timers.c:22
0x55c0fc6e70a9 main
	lightningd/lightningd.c:1201
0x7f6917006082 __libc_start_main
	../csu/libc-start.c:308
0x55c0fc6c90ad ???
	???:0
0xffffffffffffffff ???
	???:0
2023-02-10T21:48:45.394Z **BROKEN** lightningd: FATAL SIGNAL 6 (version 22.11.1)
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: common/daemon.c:38 (send_backtrace) 0x55c0fc72c13b
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: common/daemon.c:46 (crashdump) 0x55c0fc72c18d
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/x86_64/sigaction.c:0 ((null)) 0x7f691702508f
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: ../sysdeps/unix/sysv/linux/raise.c:51 (__GI_raise) 0x7f691702500b
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:79 (__GI_abort) 0x7f6917004858
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: lightningd/log.c:854 (fatal_vfmt) 0x55c0fc6e94c5
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: wallet/wallet.c:61 (db_fatal) 0x55c0fc716ba4
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: db/exec.c:114 (db_data_version_incr) 0x55c0fc74131d
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: db/exec.c:151 (db_commit_transaction) 0x55c0fc7415bb
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: lightningd/subd.c:585 (sd_msg_read) 0x55c0fc70bda2
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:59 (next_plan) 0x55c0fc7d2146
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:407 (do_plan) 0x55c0fc7d2613
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:417 (io_ready) 0x55c0fc7d26b0
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: ccan/ccan/io/poll.c:453 (io_loop) 0x55c0fc7d38ec
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: lightningd/io_loop_with_timers.c:22 (io_loop_with_timers) 0x55c0fc6e2dba
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: lightningd/lightningd.c:1201 (main) 0x55c0fc6e70a9
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: ../csu/libc-start.c:308 (__libc_start_main) 0x7f6917006082
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0x55c0fc6c90ad
2023-02-10T21:48:45.394Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0xffffffffffffffff
Log dumped in crash.log.20230210214845
lightning_channeld: Could not set sync read from master: Connection reset by peer (version 22.11.1)
Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at afterWriteDispatched (node:internal/stream_base_commons:160:15)
    at writeGeneric (node:internal/stream_base_commons:151:3)
    at Socket._writeGeneric (node:net:905:11)
    at Socket._write (node:net:917:8)
    at writeOrBuffer (node:internal/streams/writable:391:12)
    at _write (node:internal/streams/writable:332:10)
    at Socket.Writable.write (node:internal/streams/writable:336:10)
    at Plugin._write (/home/go/c-lightning-REST/node_modules/clightningjs/src/plugin.js:40:25)
    at Plugin._writeJsonrpcNotification (/home/go/c-lightning-REST/node_modules/clightningjs/src/plugin.js:105:16)
    at /home/go/c-lightning-REST/node_modules/clightningjs/src/plugin.js:164:14
Emitted 'error' event on Socket instance at:
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -32,
  code: 'EPIPE',
  syscall: 'write'
}
Aborted (core dumped)

```

**Comment by BTCBellyButton** [comment]
No improvement. I woke up this morning to the following:
![image](https://user-images.githubusercontent.com/68573637/218257629-55188702-7590-4ec3-aa74-756485efb0a2.png)

![image](https://user-images.githubusercontent.com/68573637/218257649-91e075e8-55bc-4c0d-bc9a-e5e19669ea74.png)

![image](https://user-images.githubusercontent.com/68573637/218257664-1ed45606-7bf2-4e4a-aa56-3f70b9743c22.png)

![image](https://user-images.githubusercontent.com/68573637/218257732-7e54c153-b990-429b-86da-91310118a0d6.png)


Added a 3rd drive and ran a check on the RAID1 afterwards. All clean.

Disabling WAL now.

**Comment by grubles** [comment]
Did you have WAL enabled before? In my experience, it causes its own crashes so I would keep the sqlite db defaults.

**Comment by BTCBellyButton** [comment]
> Did you have WAL enabled before? In my experience, it causes its own crashes so I would keep the sqlite db defaults.

It wasn't enabled before upgrading the DB with the new v22.11.1.
I saw your comment on another issue in the past about that and I double checked if it was there in the backup I did before the upgrade. It was not. 
So I waited to disable it  until now bc I wasn't sure it was "safe" with the new release.

For now it is running fine. I'll try loading the 2 plugins I need again and see how it goes.

So if that's the issue why was it enabled during the upgrade?

**Comment by BTCBellyButton** [comment]
If it's enabled during the DB upgrading process, you should make sure it is not or have a note somewhere. 
Especially bc there's no going back after upgrading the DB.
Mine isn't a very large node but I still have few thousands dollars in my channels so the last 2 days have been a bit stressful.

**Comment by cdecker** [comment]
Not quite sure about why WAL would be an issue here. My previous comment was led astray by the strange mount point, which seems to be mounted from somewhere else? It is possible that WAL doesn't play nicely with file systems that are not quite posix-compliant (NFS or FAT32?), which may be the real cause here. Also I wonder if either of those plugin's directly talk to the database instead of the API, in which case we might have found the culprit.

Fwiw the optimistic locking is very conservative and it triggering almost always means there's another instance running (manually started one and systemd started one?)

**Comment by BTCBellyButton** [comment]
RAID1 with mdadm. No NFS nor FAT32.
Ubuntu 20.04.
Tha author of Circular confirmed the plugin doesn't access the DB directly. Plus CLN kept crashing with no plugins running. 
CLN is stable since I disabled WAL yesterday.

**Comment by BTCBellyButton** [comment]
Confirm that after disabing WAL not one crash has occurred.

**Comment by BTCBellyButton** [comment]
Now it is 12 consecutive days that it has been running without any crashes once WAL has been disabled.
